src/common/RefCountedObj.h

```diff
@@ -193,6 +193,9 @@ static inline void intrusive_ptr_add_ref(const RefCountedObject *p) {
 static inline void intrusive_ptr_release(const RefCountedObject *p) {
   p->put();
 }
+static inline void unique_ptr_release(const RefCountedObject *p) {
+  p->put();
```

Comment:
It might be worth a note on how this relates to the `std::make_unique`. The intuition suggests to look for `get()` during the creation but, as we know, it's not there – we initialize the counter with `1` in parent's ctor.